### PR TITLE
Allow ForkingTaskRunner javaOpts to have quoted arguments which contain spaces

### PR DESCRIPTION
**Pull request by drcrallen**
It is not currently possible to pass java options to the peons for items like `-XX:OnOutOfMemoryError="kill -9 %p"` this PR is intended to allow such options to be passed to the peons.

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java [hunk 1]
```diff
@@ -457,3 +458,78 @@ private void registerWithCloser(Closer closer)
...
+        return next();
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException("remove");
+      }
+    };
+  }
+}
```
Comment (himanshug) [inline indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java]: came across http://stackoverflow.com/a/19946701 , seems simpler ... see if it makes sense to use the CharMatcher.

Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java]: Tweaked with it a tiny bit, and changed some of the unit tests. Overall I think it is cleaner.

Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java]: Thanks!

File: indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ForkingTaskRunnerTest
+{
+  @Test(expected = AssertionError.class)
+  public void testPatternMatcherFailureForJavaOptions()
+  {
+    checkValues(new String[]{"not quoted has space"});
+  }
```
Comment (himanshug) [inline indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java]: it is non trivial to understand why this throws AssertionError and testPatternMatcherPreservesNonBreakingSpacesJavaOptions() does not. can you pls add some more comments?

Comment (drcrallen) [inline indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java]: Will add comments.

Comment (drcrallen) [inline indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java]: Hopefully more clear now.

File: indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ForkingTaskRunnerTest
+{
+  // This tests the test to make sure teh test fails when it should.
```
Comment (xvrl) [inline indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java]: typo

Comment (drcrallen) [inline indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java]: fixed

**Comment by xvrl** [comment]
:+1:

**Comment by drcrallen** [comment]
`RemoteTaskRunnerTest.testWorkerRemoved:370 null `

**Pull request merged**